src/Product/ProductHelper.php

```diff
@@ -59,6 +59,7 @@ public function __construct( ProductMetaHandler $meta_handler, WC $wc, MerchantC
 	 * @param GoogleProduct $google_product
 	 */
 	public function mark_as_synced( WC_Product $product, GoogleProduct $google_product ) {
+		$this->meta_handler->delete_failed_delete_attempts( $product );
```

Comment:
💅

`delete_failed_delete_attempts` is not about `marking_as_synced` so, maybe, must be moved into its own cleanup delete attempts method and used next to `marking_as_synced` and not inside it? Same thoughts i have on `update_empty_visibility`, which is few lines below.

Or... rename `marking_as_synced` into `cleanup_delete_attempts_mark_as_synced_and_update_empty_visibility` ))

Reply:
I agree that the cleanup is not part of the syncing, however `mark_as_synced` is called from multiple locations, so we would need a wrapper around that function to call both the mark as sync and the cleanup. I'd sooner rename the function to describe it handles two actions. Although in this case, refactoring that might require quite a bit of additional testing, so I think it's better to update the function comment, to describe what actions it's performing. I'll go ahead and add that.